### ISSUE TITLE
Smart backspace for Python

### ISSUE THREAD

**Issue by defaultxr** [hrehfeld/emacs-smart-hungry-delete#5]
Not sure if there is a better/suggested way of doing this, but I wrote a simple function for backspacing in Python. Normally `smart-hungry-delete-backward-char` will delete all indentation on a line if called with the point directly after the indentation, but I usually just want it to decrease the indentation by one level. This function does that:

```emacs-lisp
(defun python-smart-backspace (&optional arg)
  "Call `smart-hungry-delete-backward-char' unless point is directly after the indentation, in which case call `python-indent-dedent-line-backspace'."
  (interactive)
  (if (looking-back "^[\t ]+")
      (python-indent-dedent-line-backspace arg)
    (smart-hungry-delete-backward-char arg)))
```

I just bind it to `<backspace>` in `python-mode-map`.

**Comment by hrehfeld** [comment]
hmmm. The correct way would be to check if point is before or at `beginning-of-indentation`, and then dedent. 

However, I have trouble finding a mode-agnostic way to dedent. We currently have no mode specific stuff other than overriding local variables. This change would require special behavior for the `check` function, and some way to signal that the fallback command should change -- in python-mode only.

**Comment by hrehfeld** [comment]
fixed in 16c5f01586177726cb32fea50efa4d54bf95641b
